utils/changelog_updater.py

```diff
@@ -18,82 +16,50 @@ def main() -> None:
     """
     Update CHANGELOG.md when API generator produces new code differing from existing.
     """
-    root_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
-
-    after_paths = [
-        os.path.join(root_dir, f"opensearchpy/{folder}")
-        for folder in ["client", "_async/client"]
-    ]
-
-    before_paths = [
-        os.path.join(root_dir, f"before_generate/{folder}")
-        for folder in ["client", "async_client"]
-    ]
-
-    # Compare only .py files and take their union for client and async_client directories
-    before_files_client = set(
-        file for file in os.listdir(before_paths[0]) if file.endswith(".py")
-    )
-    after_files_client = set(
-        file for file in os.listdir(after_paths[0]) if file.endswith(".py")
-    )
-
-    before_files_async_client = set(
-        file for file in os.listdir(before_paths[1]) if file.endswith(".py")
-    )
-    after_files_async_client = set(
-        file for file in os.listdir(after_paths[1]) if file.endswith(".py")
-    )
-
-    all_files_union_client = before_files_client.union(after_files_client)
-    all_files_union_async_client = before_files_async_client.union(
-        after_files_async_client
-    )
-
-    # Compare files and check for mismatches or errors for client and async_client directories
-    mismatch_client, errors_client = filecmp.cmpfiles(
-        before_paths[0], after_paths[0], all_files_union_client, shallow=True
-    )[1:]
-    mismatch_async_client, errors_async_client = filecmp.cmpfiles(
-        before_paths[1], after_paths[1], all_files_union_async_client, shallow=True
-    )[1:]
-
-    if mismatch_client or errors_client or mismatch_async_client or errors_async_client:
-        print("Changes detected")
-        response = requests.get(
-            "https://api.github.com/repos/opensearch-project/opensearch-api-specification/commits"
+    git_command = "git status"
+    try:
+        git_status = subprocess.check_output(
```

Comment:
For a future improvement I recommend switching to a library, e.g. https://github.com/gitpython-developers/GitPython. That should have proper methods/APIs for things like number of files to be committed or files untracked, making it a lot less brittle.